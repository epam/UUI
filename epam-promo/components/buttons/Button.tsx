import * as uui from '@epam/uui';
import { createSkinComponent, devLogger } from '@epam/uui-core';
import { FillStyle } from '../types';
import css from './Button.module.scss';

/** Defines component color. */
type ButtonColor = 'blue' | 'green' | 'red' | 'gray50' | 'gray' | uui.ButtonProps['color'];

type ButtonMods = {
    /**
     * Defines component fill style.
     * @default 'solid'
     */
    fill?: FillStyle;
    /**
     * Defines component color.
     * @default 'blue'
     */
    color?: ButtonColor;
    /**
     * Defines component size.
     * @default '36'
     */
    size?: uui.ButtonProps['size'];
};

const mapFill: Record<FillStyle, uui.ButtonProps['fill']> = {
    solid: 'solid',
    white: 'outline',
    light: 'ghost',
    none: 'none',
};

/** Represents the properties of a Button component. */
export type ButtonProps = uui.ButtonCoreProps & ButtonMods;

export const Button = /* @__PURE__ */createSkinComponent<uui.ButtonProps, ButtonProps>(
    uui.Button as any, // TODO: remove it when BaseButton inheritance will be reworked
    (props) => {
        if (__DEV__) {
            devLogger.warnAboutDeprecatedPropValue<ButtonProps, 'color'>({
                component: 'Button',
                propName: 'color',
                propValue: props.color,
                propValueUseInstead: 'gray',
                condition: () => ['gray50'].indexOf(props.color) !== -1,
            });
        }
        return {
            fill: mapFill[props.fill] || mapFill.solid,
        } as any; // TODO: need new helper to rewrite types
    },
    (props) => [
        ['42', '48'].includes(props.size) && css.uppercase,
    ],
);
