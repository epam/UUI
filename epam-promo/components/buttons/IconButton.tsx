import { createSkinComponent } from '@epam/uui-core';
import { EpamPrimaryColor } from '../types';
import * as uui from '@epam/uui';

interface IconButtonMods {
    /**
     * Defines component color.
     * @default 'gray60'
     */
    color?: EpamPrimaryColor | 'gray30' | 'gray50' | 'gray60' | uui.IconButtonProps['color'];
}

/** Represents the properties for the IconButton component. */
export type IconButtonProps = uui.IconButtonCoreProps & IconButtonMods;

export const IconButton = /* @__PURE__ */createSkinComponent<uui.IconButtonProps, IconButtonProps>(
    uui.IconButton,
    (props) => ({ color: props.color ?? 'gray60' }),
);
