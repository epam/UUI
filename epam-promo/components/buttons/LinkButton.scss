@use  '../../assets/styles/index' as *;

.root {
    &:global(.uui-button-box) {
        background-color: transparent;
        color: var(--uui-text);
        fill: var(--uui-text);

        &:global(.-clickable:not(.uui-disabled)) {
            &:hover, &:focus {
                color: var(--uui-text-hover);
                fill: var(--uui-text-hover);
            }

            &:active {
                color: var(--uui-text-active);
                fill: var(--uui-text-active);
            }

            &:focus {
                box-shadow: none;
            }
        }

        &:global(.uui-disabled) {
            color: var(--uui-text-disabled);
            fill: var(--uui-text-disabled);
            background-color: transparent;

            &:hover {
                background-color: transparent;
            }
        }
    }

    :global(.uui-caption) {
        font-family: $font-sans-semibold;
        font-weight: normal;
    }

    // Sizes
    &.size-18 {
        @include link-button-layout(18px);
        min-width: auto;
    }

    &.size-24 {
        @include link-button-layout(24px);
        min-width: auto;
    }

    &.size-30 {
        @include link-button-layout(30px);
        min-width: auto;
    }

    &.size-36 {
        @include link-button-layout(36px);
        min-width: auto;
    }

    &.size-42 {
        @include link-button-layout(42px);
        min-width: auto;

        :global(.uui-caption) {
            font-family: $font-museo-sans;
            text-transform: uppercase;
        }
    }

    &.size-48 {
        @include link-button-layout(48px);
        min-width: auto;

        :global(.uui-caption) {
            font-family: $font-museo-sans;
            text-transform: uppercase;
        }
    }
}
