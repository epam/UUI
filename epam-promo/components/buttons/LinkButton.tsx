import { createSkinComponent, devLogger } from '@epam/uui-core';
import * as uui from '@epam/uui';

type LinkButtonMods = {
    /**
     * Defines component color.
     * @default 'blue'
     */
    color?: 'blue' | 'green' | 'amber' | 'red' | 'gray60' | 'gray10' | uui.LinkButtonProps['color'];
};

/** Represents the properties for the LinkButton component. */
export type LinkButtonProps = uui.LinkButtonCoreProps & LinkButtonMods;

export const LinkButton = /* @__PURE__ */createSkinComponent<uui.LinkButtonProps, LinkButtonProps>(
    uui.LinkButton,
    (props) => {
        if (__DEV__) {
            devLogger.warnAboutDeprecatedPropValue<LinkButtonProps, 'color'>({
                component: 'LinkButton',
                propName: 'color',
                propValue: props.color,
                condition: () => ['green', 'amber', 'red'].indexOf(props.color) !== -1,
            });
        }
        return {
            color: props.color ?? 'blue',
        };
    },
);
