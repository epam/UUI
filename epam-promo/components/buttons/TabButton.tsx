import css from './TabButton.scss';
import styles from '../../assets/styles/colorvars/buttons/tabButton-colorvars.scss';
import { Button, ButtonProps } from '@epam/uui-components';
import { withMods } from '@epam/uui-core';
import { systemIcons } from '../../icons/icons';
import { getIconClass } from './helper';

export interface TabButtonMods {
    /** Tab button size */
    size?: '36' | '48' | '60';

    /** Show notification red dot */
    withNotify?: boolean;
}

function applyTabButtonMods(mods: TabButtonMods & ButtonProps) {
    return [css.root, css['size-' + (mods.size || '48')], mods.withNotify && css.uuiNotification, styles.tabButtonColor, ...getIconClass(mods)];
}

export const TabButton = withMods<ButtonProps, TabButtonMods>(Button, applyTabButtonMods, props => ({
    dropdownIcon: systemIcons['36'].foldingArrow,
    clearIcon: systemIcons['36'].clear,
    countPosition: 'right',
    ...props,
    rawProps: { role: 'tab', ...props.rawProps },
}));
