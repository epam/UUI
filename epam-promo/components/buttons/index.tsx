export * from './Button';
export * from './IconButton';
export * from './LinkButton';
export * from './TabButton';
export * from './VerticalTabButton';
