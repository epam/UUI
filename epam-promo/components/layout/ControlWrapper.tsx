import * as React from 'react';
import * as css from './ControlWrapper.scss';
import * as types from '../types';
import { IHasCX, cx } from '@epam/uui';

interface ControlWrapperProps extends IHasCX {
    size: types.ControlSize;
}

export class ControlWrapper extends React.Component<ControlWrapperProps, any> {

    render() {
        return (
            <div className={ cx(css.root, css['size-' + this.props.size], this.props.cx) }>
                { this.props.children }
            </div>
        );
    }
}