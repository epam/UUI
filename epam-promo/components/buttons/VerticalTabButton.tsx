import css from './VerticalTabButton.scss';
import { ButtonProps } from '@epam/uui-components';
import { withMods } from '@epam/uui-core';
import { TabButton, TabButtonMods } from './TabButton';

function applyVerticalTabButtonMods() {
    return [css.root];
}

export const VerticalTabButton = withMods<ButtonProps & TabButtonMods>(TabButton, applyVerticalTabButtonMods);
