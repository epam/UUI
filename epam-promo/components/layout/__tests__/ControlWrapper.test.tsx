import React from 'react';
import renderer from 'react-test-renderer';
import { ControlWrapper } from '../ControlWrapper';
import { Button } from '../../buttons';

describe('ControlWrapper', () => {
    it('should be rendered correctly', () => {
        const tree = renderer
            .create(
                <ControlWrapper size="36">
                    <Button caption="On" />
                    <Button caption="Off" color="green" />
                </ControlWrapper>
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
