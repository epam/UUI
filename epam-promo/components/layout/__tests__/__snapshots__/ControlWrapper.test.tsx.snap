// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ControlWrapper should be rendered correctly 1`] = `
<div
  className="root size-36"
>
  <button
    className="container uui-button-box uui-enabled button-vars button-primary root size-36 mode-solid uui-theme-promo"
    onClick={[Function]}
    tabIndex={-1}
    type="button"
  >
    <div
      className="uui-caption"
    >
      On
    </div>
  </button>
  <button
    className="container uui-button-box uui-enabled button-vars button-green root size-36 mode-solid uui-theme-promo"
    onClick={[Function]}
    tabIndex={-1}
    type="button"
  >
    <div
      className="uui-caption"
    >
      Off
    </div>
  </button>
</div>
`;
